// pushkin-react/es/actions/questionque.js

import local from './axiosConfigInitial';
import { requestQuestionBegin } from './fetch';
import { error } from './error';

export var ANSWER_COLLECTION = 'ANSWER_COLLECTION';
export var NEXT_QUESTION = 'NEXT_QUESTION';
export var SET_RESULTS = 'SET_RESULTS';

export function nextQuestion(nextQuestion) {
  return {
    type: NEXT_QUESTION,
    nextQuestion: nextQuestion
  };
}
function setResults(results) {
  return {
    type: SET_RESULTS,
    results: results
  };
}
function saveAnswers(answer) {
  return {
    type: ANSWER_COLLECTION,
    answer: answer
  };
}
export function postAnswerGetQuestion(response, answer) {
  return function (dispatch, getState) {
    var state = getState().pushkin;
    dispatch(requestQuestionBegin());
    if (Array.isArray(response.choiceId) && response.choiceId.length > 0) {
      return Promise.all(response.choiceId.map(function (currentId) {
        return local.post('response', {
          choiceId: currentId,
          questionId: response.questionId,
          user: {
            id: response.user.id
          }
        }).then(function (resp) {
          return resp.data;
        });
      })).then(function (data) {
        if (!data[0]) {
          return local.get('/results/' + state.userInfo.id).then(function (resp) {
            return resp.data.results;
          }).then(function (results) {
            dispatch(setResults(results));
            dispatch(nextQuestion(null));
          });
        } else {
          // if (state.options.saveAnswers) {
          //   dispatch(saveAnswers(answer))
          // }
          dispatch(nextQuestion(data[0]));
        }
      }).catch(function (err) {
        return dispatch(error(err));
      });
    }
    return local.post('response', response).then(function (resp) {
      if (resp.error) {
        return dispatch(error(resp.error));
      }
      if (!resp.data) {
        return local.get('/results/' + state.userInfo.id).then(function (resp) {
          return resp.data.results;
        }).then(function (results) {
          dispatch(setResults(results));
          dispatch(nextQuestion(null));
        });
      } else {
        // if (state.options.saveAnswers) {
        //   dispatch(saveAnswers(answer))
        // }
        dispatch(nextQuestion(resp.data));
      }
    });
    // .catch(err => {
    //   return dispatch(error(err));
    // });
  };
}
