// pushkin-react/lib/actions/questionlist.js
'use strict';

exports.__esModule = true;
exports.USER_ID = exports.BUILD_INITIAL = exports.FETCHING_LIST = undefined;
exports.questionList = questionList;

var _axiosConfigInitial = require('./axiosConfigInitial');

var _axiosConfigInitial2 = _interopRequireDefault(_axiosConfigInitial);

var _error = require('./error');

var _fetch = require('./fetch');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FETCHING_LIST = exports.FETCHING_LIST = 'FETCHING_LIST';
var BUILD_INITIAL = exports.BUILD_INITIAL = 'BUILD_INITIAL';
var USER_ID = exports.USER_ID = 'USER_ID';

function buildInitial(list) {
  console.log('buildInitial action', list);
  return {
    type: BUILD_INITIAL,
    list: list
  };
}
function sendUserId(id) {
  return {
    type: USER_ID,
    id: id
  };
}
function questionList() {
  return function (dispatch, getState) {
    console.log('questionlist');
    var action = (0, _fetch.requestQuestionBegin)();
    console.log('requestionBegin', action);
    dispatch(action);
    // console.log('getting initial questions', dispatch, getState());
    console.log('fetching');
    return _axiosConfigInitial2.default.get('initialQuestions').then(function (resp) {
      console.log('resp', resp.data);
      if (resp.error) {
        console.log('err');
        return dispatch((0, _error.error)(resp.error));
      }
      console.log('ldInitial', resp.data.questions);
      dispatch(buildInitial(resp.data.questions));
      return Promise.resolve(resp.data);
    }).then(function (data) {
      console.log('setting user id');
      return dispatch(sendUserId(data.user.id));
    }).catch(function (err) {
      console.error('ERROR', err);
      return dispatch((0, _error.error)(err));
    });
  };
}
