// pushkin-react/es/surveyprovider.js
var _class, _temp, _initialiseProps;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/* eslint-disable max-len */
import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import MultiChoice from './content/multichoice';
import MultiPicture from './content/multipicture';
import MultiSelect from './content/multiselect';
import { questionList } from './actions/questionlist';
import { postAnswerGetQuestion } from './actions/questionque';
import { startInstruction } from './actions/instruction';
import { startProgress } from './actions/progress';
import Spinner from './content/Spinner';

var SurveyProvider = (_temp = _class = function (_React$Component) {
  _inherits(SurveyProvider, _React$Component);

  function SurveyProvider(props) {
    _classCallCheck(this, SurveyProvider);

    var _this = _possibleConstructorReturn(this, _React$Component.call(this, props));

    _initialiseProps.call(_this);

    _this.state = {};
    return _this;
  }

  SurveyProvider.prototype.componentWillMount = function componentWillMount() {
    // uncomment line 30 to save answers
    // you can find your answers saved under state.questionque -> answers
    //this.props.dispatch(saveAnswers());
    if (!this.props.questionque.current && !this.props.questionque.isFetching && this.props.userInfo.results && !this.state.finishCalled) {
      // survey is done, call 
      this.setState({ finishCalled: true });
      this.props.onFinish();
    }
    this.props.dispatch(startInstruction(this.props.instructions));
  };

  SurveyProvider.prototype.componentDidMount = function componentDidMount() {
    console.log('building initial');
    // this.buildInitial();
    this.props.dispatch(questionList());
  };

  SurveyProvider.prototype.handlePictureChoices = function handlePictureChoices(currentQuestion) {
    if (currentQuestion.type === "survey-multi-picture") {
      return currentQuestion.choices.map(function (currentChoice) {
        return {
          url: currentChoice.imageUrl,
          label: currentChoice.displayText,
          choiceId: currentChoice.id
        };
      });
    }
    return currentQuestion.choices.map(function (currentChoice) {
      return currentChoice.displayText;
    });
  };

  SurveyProvider.prototype.render = function render() {
    if (this.props.questionque.isFetching) {
      return null;
    }
    if (this.props.questionque.isFetching && !this.props.questionque.current) {
      return React.createElement(
        'h3',
        null,
        'Loading ... '
      );
    }
    if (this.props.questionque.current) {
      var choices = this.handlePictureChoices(this.props.questionque.current);
      switch (this.props.questionque.current.type) {
        case "survey-multi-picture":
          {
            return React.createElement(
              'div',
              null,
              React.createElement(MultiPicture, {
                question: this.props.questionque.current.prompt,
                choices: choices,
                showProgress: false,
                dispatchPrecent: this.dispatchPrecent,
                questionId: this.props.questionque.current.choices[0].questionId,
                trialId: this.props.questionque.current.trialId,
                nextQuestion: this.fetchNextQuestion,
                progress: this.props.progress,
                userId: this.props.userInfo.id
              })
            );
          }
        case "survey-multi-choice":
          {
            return React.createElement(
              'div',
              null,
              React.createElement(MultiChoice, {
                question: this.props.questionque.current.prompt,
                choices: choices,
                showProgress: false,
                dispatchPrecent: this.dispatchPrecent,
                allChoices: this.props.questionque.current.choices,
                questionId: this.props.questionque.current.choices[0].questionId,
                trialId: this.props.questionque.current.trialId,
                nextQuestion: this.fetchNextQuestion,
                progress: this.props.progress,
                userId: this.props.userInfo.id
              })
            );
          }
        case "survey-multi-select":
          {
            return React.createElement(
              'div',
              null,
              React.createElement(MultiSelect, {
                question: this.props.questionque.current.prompt,
                choices: choices,
                showProgress: false,
                dispatchPrecent: this.dispatchPrecent,
                allChoices: this.props.questionque.current.choices,
                questionId: this.props.questionque.current.choices[0].questionId,
                trialId: this.props.questionque.current.trialId,
                nextQuestion: this.fetchNextQuestion,
                progress: this.props.progress,
                userId: this.props.userInfo.id
              })
            );
          }
        default:
          return null;
      }
    }
    if (!this.props.questionque.current && !this.props.questionque.isFetching && this.props.userInfo.results) {
      return React.createElement(
        'div',
        null,
        this.props.resultsContainer(this.props.userInfo.results)
      );
    }
    return React.createElement(
      'h1',
      null,
      'here'
    );
  };

  return SurveyProvider;
}(React.Component), _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.fetchNextQuestion = function (response, answer) {
    var props = _this2.props;
    // Disabled to allow for empty responses by users
    // if (!response.choiceId) {
    //   console.log(response, 'had no choice id');
    // } else {
    if (!response.questionId) {
      console.log(response, 'had no question id');
    } else {
      props.dispatch(postAnswerGetQuestion(response));
    }
    // }
  };

  this.dispatchPrecent = function (numberOfQuestions) {
    var precent = 100 / numberOfQuestions;
    // this.props.dispatch(startProgress((parseFloat(this.props.options.precent, 10)) + precent))
  };
}, _temp);

export default connect(function (state) {
  return state.pushkin;
})(SurveyProvider);
